### DISCUSSION TITLE
Prisma mock works in a file but doesn't in other files

### DISCUSSION THREAD

**Discussion by KhaledMAlasmari** [prisma/prisma#14229] [Q&A] [answered]
Hello
I have these two files with these two tests where I'm mocking the create function
[Clip.test.ts
](https://github.com/KhaledMAlasmari/pclip/blob/master/src/__tests__/model/Clip.test.ts)
```ts
    test('should save a clip successfully to the database', async () => {
        const clipInfo = {
            id: 1,
            title: "bussniess lmao!",
            suffix: "cl59pyr300000356k9uxnpobj",
            originalLink: "https://pca.st/wmd1zmul",
            startTimestamp: "00:30:32",
            endTimestamp: "00:31:32",
            createdAt: new Date(),
            clipOwnerId: null,
            originalPodcastId: null
        }
        prismaMock.clip.create.mockResolvedValue(clipInfo)
        await expect(Clip.createClip(clipInfo.originalLink, clipInfo.startTimestamp, clipInfo.endTimestamp, clipInfo.title)).resolves.toEqual(
            clipInfo.suffix
        )
    })
```

[User.test.ts](https://github.com/KhaledMAlasmari/pclip/blob/master/src/__tests__/model/User.test.ts)
```ts
  test('should create new user successfully', async () => {
    const password = "HE1hexd12$"
    const userInfo = {
      id: 1,
      name: 'Rich',
      email: 'sultan@gmail.com',
      hashedPassword: await User.hashPassword(password),
      createdAt: new Date(),
      isEmailVerified: true
    }

    prismaMock.user.create.mockResolvedValue(userInfo)

    await expect(User.createUser(userInfo.name, userInfo.email, password)).resolves.toEqual({
      name: 'Rich',
      email: 'sultan@gmail.com',
    })
  })

```

the mock seems to work in User.test.ts but does not in any other file. The tests inside Clip.test.ts connects to the database and create records. When I moved the tests to the User.test.ts file and it worked just fine
[The repo link](https://github.com/KhaledMAlasmari/pclip)

Thank you in advance!

**Comment by KhaledMAlasmari** [comment, marked as answer]
Whoops, Apparently I removed
```js
  setupFilesAfterEnv: ['./src/__tests__/mock/singleton.ts']
```
from jest.config.js which broke everything!
Thank you anyways!

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
